src/TimePicker/TimePicker.tsx

```diff
@@ -95,28 +102,28 @@ const maybeInsertSeparator = str => {
   return str;
 };
 
-export const parseInTimeFormat = (inputStr, timeFormat) => {
+export const parseInTimeFormat = (inputStr: string, timeFormat: TimeFormat): OnChangeInput => {
   if (inputStr === '') {
     return { originalInput: inputStr };
   }
   const cleanedInput = maybeAddTrailingZero(maybeInsertSeparator(cleanSeparator(inputStr)));
   const inputArray = cleanedInput.split(separator);
   const [_hours, _minutes] = inputArray;
-  const hours = numberRegex.test(_hours) ? parseInt(_hours, 10) : '';
-  const minutes = !_minutes || numberRegex.test(_minutes) ? parseInt(_minutes, 10) : '';
+  const hours = numberRegex.test(_hours) ? parseInt(_hours, 10) : null;
+  const minutes = !_minutes || numberRegex.test(_minutes) ? parseInt(_minutes, 10) : null;
   const validHours = isValidHoursInTimeFormat(hours, timeFormat);
   const validMinutes = Minute.is(minutes);
 
   if (validHours && validMinutes) {
-    return { hours, minutes, originalInput: cleanedInput };
+    return { hours: hours!, minutes: minutes!, originalInput: cleanedInput };
```

Comment:
are these safe?

Reply:
yep. They're checked in `isValidHoursInTimeFormat` and `Minute.is(minutes)`

Reply:
mm, then we shouldn't need this if we use TS properly. I'll give it a look